packages/navigator/src/browser/navigator-widget.tsx

```diff
@@ -172,22 +173,41 @@ export class FileNavigatorWidget extends FileTreeWidget {
         }
     }
 
+    protected canOpenWorkspaceFileAndFolder() {
+        return isOSX || !environment.electron.is();
+    }
+
     protected readonly openWorkspace = () => this.doOpenWorkspace();
     protected doOpenWorkspace() {
         this.commandService.executeCommand(WorkspaceCommands.OPEN_WORKSPACE.id);
     }
 
+    protected readonly openFolder = () => this.doOpenFolder();
+    protected doOpenFolder() {
+        this.commandService.executeCommand(WorkspaceCommands.OPEN_FOLDER.id);
+    }
+
     /**
      * Instead of rendering the file resources from the workspace, we render a placeholder
      * button when the workspace root is not yet set.
      */
     protected renderOpenWorkspaceDiv(): React.ReactNode {
+        const openWorkspaceDescription = this.canOpenWorkspaceFileAndFolder() ?
+            'Select a folder or a workspace-file to open as your workspace' :
+            'Select a workspace-file to open as your workspace';
+        const openWorkspace = <button className='open-workspace-button' title={openWorkspaceDescription} onClick={this.openWorkspace}>
+            Open Workspace
+        </button>;
+
+        const openFolder = <button className='open-workspace-button' title='Select a folder as your workspace root' onClick={this.openWorkspace}>
+            Open Folder
+        </button>;
+
         return <div className='theia-navigator-container'>
             <div className='center'>You have not yet opened a workspace.</div>
             <div className='open-workspace-button-container'>
-                <button className='open-workspace-button' title='Select a folder as your workspace root' onClick={this.openWorkspace}>
-                    Open Workspace
-                </button>
+                {openWorkspace}
+                {!this.canOpenWorkspaceFileAndFolder() ? openFolder : ''}
```

Comment:
should we instead keep a global boolean variable for `isOSX || !environment.electron.is()` so we reduce unnecessary calls ?